BibBuddy/app/src/main/java/de/bibbuddy/BookOnlineFragment.java

```diff
@@ -91,25 +92,32 @@ private void closeFragmentAfterAdding() {
   }
 
   private void handleIsbnInput(int keyCode, KeyEvent event) {
+    String textInput = searchFieldText.getText().toString().replaceAll("\\s", "");
+
     if (keyCode == KeyEvent.KEYCODE_ENTER) {
       if (event.getAction() != KeyEvent.ACTION_DOWN) {
-        isbnRetriever = new IsbnRetriever(searchFieldText.getText().toString());
-        thread = new Thread(isbnRetriever);
-        thread.start();
-
-        try {
-          thread.join();
-        } catch (Exception e) {
-          System.out.println(e);
-        }
-
-        // retrieve metadata that was saved
-        Book book = isbnRetriever.getBook();
-        List<Author> authors = isbnRetriever.getAuthors();
-        if (book != null) {
-          handleAddBook(book, authors);
+        if (dataValidation.isValidIsbn10or13(textInput)) {
```

Comment:
```suggestion
        if (DataValidation.isValidIsbn10or13(textInput)) {
```
Access static method this way (without using separate instance of DataValidation.

Reply:
Sure, didn't think of this - thanks!